plugins/quetz_repodata_zchunk/quetz_repodata_zchunk/main.py

```diff
@@ -0,0 +1,102 @@
+import bz2
+import hashlib
+import json
+import os
+import subprocess
+import tempfile
+from datetime import datetime, timezone
+
+import quetz
+from quetz.tasks.indexing import _jinjaenv
+
+
+def update_index(pkgstore, updated_files, channel_name, subdir, packages):
```

Comment:
we discussed with @madhur-tandon  to  re-implement the update index function in the main tree using the new pkgstore.list_files function. This would avoid re-doing the index in each plugin. @madhur-tandon did you try this?

Reply:
It's done with the other way of using a `files` dict where each key is a `subdir` and the corresponding `value` is a list of all the files associated with the `subdir`. It's better than `pkgstore.list_files` because:
1) `pkgstore.list_files` gives us all the filenames and not their contents. So we will first have to filter which files do we really want (according to `subdir`).
2) Then, since it only returns filenames, we would need to extract raw bytes data out of each file to calculate the `sha256` and `md5` sum. But, extraction of raw bytes is different for different types of files (`.json`, `.json.bz2`, `.tar.bz2`, `.map`, etc.). Further, we need not do this for packages (`.tar.bz2`) since their hashes are already there in `repodata.json`. So it gets more complicated.

Thus, a very simple straightforward thing to do is to implement the `post_package_indexing` hook and make a function call to the `add_entry_for_index` function. Just like it's done here: https://github.com/mamba-org/quetz/pull/185/files#diff-bc7de45aaa4b34886df14bb6c346ed1e79c5aaf3bc3655f36fc5de9fd9885297R36